cmd/all-in-one/main.go

```diff
@@ -103,7 +100,10 @@ by default uses only in-memory database.`,
 				svc.MetricsFactory.Namespace(metrics.NSOptions{Name: "jaeger"}))
 			version.NewInfoMetrics(metricsFactory)
 
-			tracerCloser := initTracer(svc)
+			tracer, err := jtracer.New("jaeger-all-in-one")
+			if err != nil {
+				logger.Fatal("Failed to initialize tracer", zap.Error(err))
```

Comment:
Hmmm... I wonder why we don't annotate and return the error here instead of logging it.

Reply:
I think mostly because if we return an error, it is intercepted at the top level of main like this:
```
	if err := command.Execute(); err != nil {
		log.Fatal(err)
	}
```
Here the zapp logger is not available so the stdlib `log` is used, which is formatted differently. E.g.:
```shell
$ go run ./cmd/all-in-one --admin.http.tls.enabled=true --admin.http.tls.key=invalid
. . .
2023/07/20 11:43:08 cannot initialize admin server: failed to watch key pair invalid and : open invalid: no such file or directory
exit status 1
```
With zapp logs you get line number and other nice things.

Reply:
That's cool thing I noticed here. 🏷️